modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java

```diff
@@ -4205,12 +4121,40 @@ private void executeApplicationFlow(String flowName) throws MuleException {
         .build());
   }
 
-  private void assertZombieApplication(String appId) {
-    final Map<URI, Long> zombieMap = deploymentService.getZombieApplications();
-    assertEquals("Wrong number of zombie apps registered.", 1, zombieMap.size());
-    final Map.Entry<URI, Long> zombie = zombieMap.entrySet().iterator().next();
-    assertEquals("Wrong URL tagged as zombie.", appId, new File(zombie.getKey()).getParentFile().getName());
-    assertTrue("Invalid lastModified value for file URL.", zombie.getValue() != -1);
+  private void assertZombieApplication(String appId, int expectedZombieCount, boolean compareAgainstParentFileName) {
```

Comment:
Is very hard to understand how to invoke this method. 
ExpectdZombieCount always receives a 1 or 0, so is basically a boolean.
compareAgainstParentFileName is another boolean, but when must be true/false?
Make the four possible combination of parameter clear in a method name (ie: add method for all of them). 
You can move all these utility functions to a separated class, so you can make visible only the new methods and keep using the old method internally

Reply:
The idea was to be able to check whether an artifact was a zombie, even if it was not the only one. Maybe in the future, we added another test where the number of zombies was bigger than 1 and we still can use this method. As for now, you are right. I will leave it as it was, just checking for 1 zombie present.

Reply:
If you want to check for multiple zombies, you will need to pass multiple artifactId and multiple compareAgainstParentFileName values